cmd/execution/main.go

```diff
@@ -377,7 +380,11 @@ func main() {
 		}).
 		Component("receipt provider engine", func(node *cmd.FlowNodeBuilder) (module.ReadyDoneAware, error) {
 			retrieve := func(blockID flow.Identifier) (flow.Entity, error) {
```

Comment:
can we replace this and use (?)
```suggestion
				retrieve := myReceipts.MyReceipt
```

there is still some mismatch as `MyReceipt` returns `*flow.ExecutionReceipt`, while `RetrieveFunc` returns an `flow.Entity`.

My suggestion would be to just hand `myReceipts` to the execution Node's `provider.Engine`